scripts/test/Muon/fitting_widgets/model_fitting/model_fitting_presenter_test.py

```diff
@@ -0,0 +1,325 @@
+        self.view.set_datasets_in_function_browser.assert_called_once_with(self.dataset_names)
+        self.view.update_dataset_name_combo_box.assert_called_once_with(self.dataset_names)
+        self.view.update_y_parameters.assert_called_once_with(self.y_parameters)
+        self.view.update_x_parameters.assert_called_once_with(self.x_parameters, emit_signal=True)
+
+    def test_that_handle_parameter_combinations_error_will_show_a_warning_in_the_view(self):
+        error = "Error message"
+        self.presenter.handle_parameter_combinations_error(error)
+
+        self.presenter.disable_editing_notifier.notify_subscribers.assert_called_once_with()
+        self.view.warning_popup.assert_called_once_with(error)
+
+    def test_that_handle_function_structure_changed_will_update_the_fit_functions_and_notify_they_are_updated(self):
+        self.presenter.update_fit_functions_in_model_from_view = mock.Mock()
+        self.presenter.automatically_update_function_name = mock.Mock()
+        self.presenter.reset_fit_status_and_chi_squared_information = mock.Mock()
+
+        self.presenter.handle_function_structure_changed()
+
+        self.presenter.update_fit_functions_in_model_from_view.assert_called_once_with()
+        self.presenter.automatically_update_function_name.assert_called_once_with()
+        self.model.get_active_fit_function.assert_called_once_with()
+        self.presenter.reset_fit_status_and_chi_squared_information.assert_called_once_with()
+        self.model.update_plot_guess.assert_called_once_with()
+        self.presenter.fit_function_changed_notifier.notify_subscribers.assert_called_once_with()
+
+    def test_that_update_dataset_names_in_view_and_model_will_update_the_datasets_in_the_model_and_view(self):
+        self.presenter.update_dataset_names_in_view_and_model()
+
+        self.model.get_workspace_names_to_display_from_context.assert_called_once_with()
+        self.mock_model_result_table_names.assert_has_calls([mock.call(self.result_table_names), mock.call()])
+        self.view.update_result_table_names.assert_called_once_with(self.result_table_names)
+
+    def test_that_update_selected_parameter_combination_workspace_calls_the_expected_methods(self):
+        self.presenter.update_selected_parameter_combination_workspace()
+
+        self.mock_model_dataset_names.assert_called_once_with()
+        self.mock_model_current_dataset_index.assert_called_once_with(0)
+        self.mock_view_current_dataset_name.assert_called_once_with(self.param_combination_name)
+
+    def test_that_clear_current_fit_function_for_undo_will_only_clear_the_current_function(self):
+        self.presenter.clear_current_fit_function_for_undo()
+
+        self.model.clear_undo_data_for_current_dataset_index.assert_called_once_with()
+        self.model.number_of_undos.assert_called_once_with()
+        self.view.set_number_of_undos.assert_called_once_with(1)
+
+    def _setup_mock_view(self):
```

Comment:
are these not better in a separate file? They look like they could be reused/shared

Reply:
Its difficult to separate these because I need the property mock objects to be part of the test class. However, I have already moved most of the commonly used mocked functions to `fitting_mock_setup.py`